src/dirs.rs

```diff
@@ -0,0 +1,57 @@
+#![allow(unused)]
+
+use directories::ProjectDirs;
+use std::path::{Path, PathBuf};
+
+/// For Operating Systems beyond Windows or UNIX. May also be used for testing
+/// and mobile platforms.
+pub struct OverRide {
+    path: PathBuf,
+}
+
+impl OverRide {
+    /// This path will be the root directory that all files will be read and written.
+    pub fn new(path: &str) -> Self {
+        Self {
+            path: PathBuf::from(path.to_owned()),
+        }
+    }
+}
+
+pub enum Dirs {
+    #[cfg(not(any(target_os = "android", target_os = "androideabi", target_os = "ios")))]
+    Desktop(ProjectDirs),
```

Comment:
Can we match the enum order to the order in the functions ? It seems all functions put `Desktop` as second - we should perhaps have it here in that order too - usually makes looking at stuff easier (and i suspect most of our code is thus)